h2o-r/tests/testdir_algos/stackedensemble/runit_stackedensemble_metalearner_transform.R

```diff
@@ -0,0 +1,63 @@
+setwd(normalizePath(dirname(R.utils::commandArgs(asValues = TRUE)$"f")))
+source("../../../scripts/h2o-r-test-setup.R")
+
+stackedensemble.metalearner_transform_works <- function() {
+  train <- h2o.uploadFile(locate("smalldata/extdata/prostate.csv"))
+  y <- "CAPSULE"
+  train[[y]] <- as.factor(train[[y]])
+  nfolds <- 5
+
+  my_rf <- h2o.randomForest(y = y,
+                            training_frame = train,
+                            ntrees = 10,
+                            nfolds = nfolds,
+                            fold_assignment = "Modulo",
+                            keep_cross_validation_predictions = TRUE,
+                            seed = 1)
+
+  my_xrf <- h2o.randomForest(y = y,
+                             training_frame = train,
+                             ntrees = 10,
+                             histogram_type = "Random",
+                             nfolds = nfolds,
+                             fold_assignment = "Modulo",
+                             keep_cross_validation_predictions = TRUE,
+                             seed = 1)
+
+  stack_notransform <- h2o.stackedEnsemble(y = y,
+                                           training_frame = train,
+                                           base_models = list(my_rf, my_xrf),
+                                           metalearner_transform = "none",
+                                           seed = 1)
+
+  stack_logit <- h2o.stackedEnsemble(y = y,
+                                     training_frame = train,
+                                     base_models = list(my_rf, my_xrf),
+                                     metalearner_transform = "logit",
+                                     seed = 1)
+
+  stack_percentile_rank <- h2o.stackedEnsemble(y = y,
+                                               training_frame = train,
+                                               base_models = list(my_rf, my_xrf),
+                                               metalearner_transform = "percentile_rank",
+                                               seed = 1)
+
+  # Check that the metalearner transform was set properly
+  expect_equal(stack_logit@parameters$metalearner_transform, "Logit")
+  expect_equal(stack_percentile_rank@parameters$metalearner_transform, "PercentileRank")
```

Comment:
@tomasfryda @ledell I think this is really strange; people shouldn't see values they cannot actually use - this will make hard re-training the model using the same parameters

Reply:
are you talking about the difference in the case?
This has always been a challenge to get both clients and backend aligned on this, already with the metrics...

Reply:
```R
histogram_type = "Random",
nfolds = nfolds,
fold_assignment = "Modulo",
```
those are not snake case besides

Reply:
> are you talking about the difference in the case?

yes

> already with the metrics

I don't think we should introduce another special case, especially not in this PR.

This can and should be addressed on a more system-level with all use cases taken into consideration.

Reply:
Ok, I agree that a more system-level approach would be better especially since we would make special logic for `metalearner_transform` in the bindings generation which basically enforces that it doesn't behave the same way as other parameters which would make it harder in future to unite the "coding style" as we would have to deal with this specific logic separately.

So I'll revert the commit. Should I keep the `NONE` or change it to `None`?

Reply:
@sebhrusen if the snake case is an issue, can we just remove PercentileRank which is not really needed anyway?

This is getting complicated for a last-minute change.

Changes like that should be kept minimal, I think that is what we agreed to do.

Reply:
I kept the `NONE` for now as it is the most frequent variant on current `master` branch.

```bash
~/sources/h2o-3/h2o-r/h2o-package/R$ grep -l None *.R | wc -l 
       8

~/sources/h2o-3/h2o-r/h2o-package/R$ grep -l none *.R | wc -l
       9

~/sources/h2o-3/h2o-r/h2o-package/R$ grep -l NONE *.R | wc -l
      14
```

Reply:
@michalkurka The backend already accepts almost anything from the client, including snake-case.
The problem here is the case that we chose to use when exposing backend values to a given client compared with the case we chose to accept from the user on that same client.
- `expect_equal(stack_logit@parameters$metalearner_transform, "Logit")` : here "Logit" is what the backed returns, using the enum name.
- so we should accept the same case on the client.

just in `stackedensemble.R` before this PR:
```R
metalearner_algorithm = c("AUTO", "deeplearning", "drf", "gbm", "glm", "naivebayes", "xgboost"),
metalearner_fold_assignment = c("AUTO", "Random", "Modulo", "Stratified"),
auc_type = c("AUTO", "NONE", "MACRO_OVR", "WEIGHTED_OVR", "MACRO_OVO", "WEIGHTED_OVO")
```
and in `automl.R`:
```R
stopping_metric = c("AUTO", "deviance", "logloss", "MSE", "RMSE", "MAE", "RMSLE", "AUC", "AUCPR", "lift_top_group", "misclassification", "mean_per_class_error"),
```
honestly, this is a complete mess.

I agree that this should be handled globally, but this is already the case on the backend (no pun intended).
This is also fine on Python, because we can just usually lowercase everything before validating.

It's mainly an R issue because of this `variable=c('default', 'choice1', 'choice2')` approach, and I don't see a way to fix this there outside one of:
1. get rid of those `c(…)` in the signature. and validate programmatically like in Py.
2. add all possible cases in those `c(…)` initializers.

Alternative: add a client param to each request, and depending on the client the REST API will format all enums use a preferred case. I can already say that I don't like this solution.

Also, afaik all recent R additions have been using snake case (not only for values but also for function names).

I don't plan to fix this here, just starting a discussion, otherwise we'll still mention this issue in the next PR.

Reply:
In R, we can have a strict name choice but then also lazily accept a lower/different case version (or any other mapping) rather than using the strict `match.args()` check. Just as an option.